src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedRootModuleType.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.CodeGen;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents the <see cref="Cci.RootModuleType"/> that will be emitted. Used for <see
+    /// cref="SynthesizedRootModuleTypeStaticConstructor.ContainingType"/> so that tests can use proper metadata names.
+    /// See <see cref="CompilationTestData.GetMethodsByName"/>.
+    /// </summary>
+    internal sealed class SynthesizedRootModuleType : NamedTypeSymbol
```

Comment:
See the commit "Fix the <Module> method full names in the CompilationTestData that the compiler produces"
This enables the test code to be `verifier.VerifyIL("<Module>..cctor",` rather than `verifier.VerifyIL(".cctor",`.

The synthesized PrivateImplementationDetails methods use `null` for `ContainingType`, so they have incorrect metadata names in CompilationTestData:

https://github.com/dotnet/roslyn/blob/db5c01e59a0fea97281a1adde921c097f48e73e6/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedGlobalMethodSymbol.cs#L74-L89